lib/spack/spack/cmd/install.py

```diff
@@ -237,36 +237,39 @@ def install(parser, args, **kwargs):
     reporter.specs = specs
     with reporter:
         if args.overwrite:
-            # If we asked to overwrite an existing spec we must ensure that:
-            # 1. We have only one spec
-            # 2. The spec is already installed
-            assert len(specs) == 1, \
-                "only one spec is allowed when overwriting an installation"
-
-            spec = specs[0]
-            t = spack.store.db.query(spec)
-            assert len(t) == 1, "to overwrite a spec you must install it first"
-
-            # Give the user a last chance to think about overwriting an already
-            # existing installation
-            if not args.yes_to_all:
-                tty.msg('The following package will be reinstalled:\n')
-
-                display_args = {
-                    'long': True,
-                    'show_flags': True,
-                    'variants': True
-                }
-
-                spack.cmd.display_specs(t, **display_args)
-                answer = tty.get_yes_or_no(
-                    'Do you want to proceed?', default=False
-                )
-                if not answer:
-                    tty.die('Reinstallation aborted.')
-
-            with fs.replace_directory_transaction(specs[0].prefix):
-                install_spec(args, kwargs, specs[0])
+
+            for spec in specs:
+                installed = spack.store.db.query(spec)
+                assert len(installed) <= 1, \
+                    "Error: Multiple installed specs match " \
+                    "concrete spec %s" % spec
+
+                if not args.yes_to_all:
```

Comment:
In case of multiple specs given, the prompt below seems to require more than one user interaction (and they are interleaved with installations). Wouldn't it be better, if we want to generalize this, to display a summary of what will happen and ask for a single answer?